theano/scalar/basic.py

```diff
@@ -2150,6 +2150,8 @@ def grad(self, inputs, gout):
             else:
                 return [x.zeros_like()]
 
+        if x.dtype.startswith('float'):
+            return gz * sgn(x),
```

Comment:
That looks like a more stable formula, indeed.
Bud could you check for `x in float_dtypes` instead?